src/views/Pools/components/PoolCard/Modals/StakeModal.tsx

```diff
@@ -91,9 +92,15 @@ const StakeModal: React.FC<StakeModalProps> = ({
   const getTokenLink = stakingToken.address ? `/swap?outputCurrency=${getAddress(stakingToken.address)}` : '/swap'
 
   useEffect(() => {
-    if (stakingLimit.gt(0) && !isRemovingStake) {
-      const fullDecimalStakeAmount = getDecimalAmount(new BigNumber(stakeAmount), stakingToken.decimals)
-      setHasReachedStakedLimit(fullDecimalStakeAmount.plus(userData.stakedBalance).gt(stakingLimit))
+    const fullDecimalStakeAmount = getDecimalAmount(new BigNumber(stakeAmount), stakingToken.decimals)
+
+    if (isRemovingStake) {
+      setUserNotEnoughToken(userData.stakedBalance.lt(fullDecimalStakeAmount))
+    } else {
+      if (stakingLimit.gt(0)) {
+        setHasReachedStakedLimit(fullDecimalStakeAmount.plus(userData.stakedBalance).gt(stakingLimit))
+      }
+      setUserNotEnoughToken(userData.stakingTokenBalance.lt(fullDecimalStakeAmount))
```

Comment:
I don't think we need a new state object for this. We can simply calculate whether the value entered is greater than the balance in each render.

Reply:
Done